src/NServiceBus.AcceptanceTests/Timeout/TemporarilyUnavailableTimeoutPersister.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace NServiceBus.AcceptanceTests.Timeouts
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using NServiceBus.Extensibility;
+    using Timeout.Core;
+
+    class TemporarilyUnavailableTimeoutPersister : IPersistTimeouts, IQueryTimeouts
+    {
+        private int secondsToWait;
+
+        public int SecondsToWait
+        {
+            get { return secondsToWait; }
+            set
+            {
+                secondsToWait = value;
+                NextChangeTime = DateTime.Now.AddSeconds(SecondsToWait);
+            }
+        }
+
+        static bool isAvailable = false;
+        Task completedTask = Task.FromResult(0);
+        DateTime NextChangeTime;
+        Dictionary<string, TimeoutData> storage = new Dictionary<string, TimeoutData>(); 
+
+        private void ThrowExceptionUntilWait()
```

Comment:
It's not random though, it is:
throe exception for every call for ten seconds, then no exceptions for ten seconds, repeat